src/org/openstreetmap/josm/gui/tagging/presets/TaggingPresetSelector.java

```diff
@@ -279,7 +292,7 @@ public static class PresetClassifications implements Iterable<PresetClassificati
 
         private final List<PresetClassification> classifications = new ArrayList<>();
 
-        public List<PresetClassification> getMatchingPresets(String searchText, boolean onlyApplicable, boolean inTags,
+        public List<PresetClassification> getMatchingPresets(String searchText, boolean onlyApplicable, boolean inTags, boolean isDeprecated,
```

Comment:
We should probably use an enum here. We could probably reuse the one I mentioned earlier (so `getMatchingPresets(String searchText, DisplayCheckboxes...`, so `DisplayCheckboxes` is probably the wrong name; see if you can find one yourself).

Reply:
how about `SearchStringFilters`?

Reply:
That would probably work. But we have other locations where we have search strings. It might be better to have some more identifying information. Although this really depends upon where we put the `enum` -- if it is in its own file, it will need a much better identifier, such as `PresetSearchFilter`. If it remains in the `TaggingPresetSelector` class, then that would be part of the identifier (`TaggingPresetSelector.SearchStringFilters.DEPRECATED`).